packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,135 @@ def apply_default_cascade(args):
+                "No data were provided. Please provide data either with the `data_frame` or with the `dimensions` argument."
+            )
+        else:
+            df_args = args["data_frame"]
+            df[df_args.columns] = df_args[df_args.columns]
+
+    # Valid column names
+    df_columns = (
+        args["data_frame"].columns
+        if isinstance(args["data_frame"], pd.DataFrame)
+        else None
+    )
+    # Loop over possible arguments
+    for field_name in attrables:
+        argument_list = (
+            [args.get(field_name)]
+            if field_name not in array_attrables
+            else args.get(field_name)
+        )
+        # argument not specified, continue
+        if argument_list is None or argument_list is [None]:
+            continue
+        # Argument name: field_name if the argument is a list
+        # Else we give names like ["hover_data_0, hover_data_1"] etc.
+        field_list = (
+            [field_name]
+            if field_name not in array_attrables
+            else [field_name + "_" + str(i) for i in range(len(argument_list))]
+        )
+        # argument_list and field_list ready, iterate over them
+        for i, (argument, field) in enumerate(zip(argument_list, field_list)):
+            length = len(df)
+            if argument is None:
+                continue
+            elif isinstance(argument, str) or isinstance(
+                argument, int
+            ):  # just a column name
+                if not isinstance(
+                    args.get("data_frame"), pd.DataFrame
+                ) and not isinstance(args.get("data_frame"), np.ndarray):
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument."
+                    )
+                # Check validity of column name
+                if df_columns is not None and argument not in df_columns:
+                    raise ValueError(
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(df_columns)), argument)
+                    )
+                if length and len(args["data_frame"][argument]) != length:
+                    raise ValueError("All arguments should have the same length.")
+                df[str(argument)] = args["data_frame"][argument]
+                if isinstance(argument, int):
+                    if field_name not in array_attrables:
+                        args[field_name] = str(argument)
+                    else:
+                        args[field_name][i] = str(argument)
+                continue
+            # Case of index
+            elif isinstance(argument, pd.core.indexes.multi.MultiIndex):
+                raise TypeError("pandas MultiIndex is not supported by plotly express")
```

Comment:
for all these errors we should clearly indicate which attribute is causing the issue.